src/d3a/models/area/__init__.py

```diff
@@ -115,6 +121,35 @@ def __init__(self, name: str = None, children: List["Area"] = None,
         self.redis_ext_conn = RedisMarketExternalConnection(self) \
             if external_connection_available is True else None
 
+    def area_reconfigure_event(self, **kwargs):
+        assert all(k in self.reconfig_parameters for k in kwargs.keys())
+        if key_in_dict_and_not_none(kwargs, 'transfer_fee_const') or \
+                key_in_dict_and_not_none(kwargs, 'grid_fee_percentage'):
+            transfer_fee_const = kwargs["transfer_fee_const"] \
+                if key_in_dict_and_not_none(kwargs, 'transfer_fee_const') else None
+            grid_fee_percentage = kwargs["grid_fee_percentage"] \
+                if key_in_dict_and_not_none(kwargs, 'grid_fee_percentage') else None
+            validate_area(grid_fee_percentage=grid_fee_percentage,
+                          grid_fee_constant=transfer_fee_const)
+            self._set_grid_fees(transfer_fee_const, grid_fee_percentage)
+
+        if key_in_dict_and_not_none(kwargs, 'baseline_peak_energy_import_kWh'):
+            self.baseline_peak_energy_import_kWh = kwargs['baseline_peak_energy_import_kWh']
+            validate_area(baseline_peak_energy_import_kWh=self.baseline_peak_energy_import_kWh)
+        if key_in_dict_and_not_none(kwargs, 'baseline_peak_energy_export_kWh'):
+            self.baseline_peak_energy_export_kWh = kwargs['baseline_peak_energy_export_kWh']
+            validate_area(baseline_peak_energy_export_kWh=self.baseline_peak_energy_export_kWh)
+
+        if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA') or \
+                key_in_dict_and_not_none(kwargs, 'export_capacity_kVA'):
+            import_capacity_kVA = kwargs["import_capacity_kVA"] \
+                if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA') else None
+            export_capacity_kVA = kwargs["export_capacity_kVA"] \
+                if key_in_dict_and_not_none(kwargs, 'export_capacity_kVA') else None
+            validate_area(import_capacity_kVA=import_capacity_kVA,
+                          export_capacity_kVA=export_capacity_kVA)
+            self._convert_area_throughput_kva_to_kwh(import_capacity_kVA, export_capacity_kVA)
```

Comment:
The problem that I see with this parameter is that it is not saved to the market, but only in the Area object. This will have the effect that the event will update this parameter immediately and not from the next market slot. I am not sure if this is in the context of this story though, if it is not I would leave a comment to deal with it in the context of https://gridsingularity.atlassian.net/browse/D3ASIM-2332

Reply:
You are right. it will be updated immediately, which is not wanted.
I guess we only need to trigger this function at the beginning of the market slot, indeed in your mentioned story. Thanks!